shared/engine/transport-shared.js

```diff
@@ -2,113 +2,90 @@
     } else {
       once = true
-      f(...args)
-    }
-  }
-}
 
-// Wrapped to add logging
-function _makeLogged<F: Function>(
-  f: F,
-  type: rpcLogType,
-  logTitle: string,
-  extraInfo?: ?Object,
-  titleFromArgs?: ?Function
-): F {
-  if (printRPC) {
-    // $FlowIssue
-    return (...args) => {
-      rpcLog(type, titleFromArgs ? titleFromArgs(...args) : logTitle, {...extraInfo, args})
-      f(...args)
-    }
-  } else {
-    return f
-  }
-}
+      if (printRPC) {
+        rpcLog({
+          extra,
+          method: methodInCallback ? a1.method : method || 'unknown',
+          type: type,
+        })
+      }
+
+      if (printRPCStats) {
+        const m = methodInCallback ? a1.method : method || 'unknown'
 
-// Wrapped to make time delayed functions to test timing issues
-function _makeDelayed<F: Function>(f: F, amount: number): F {
-  if (__DEV__ && amount > 0) {
-    // $FlowIssue
-    return (...args) => {
-      localLog('%c[RPC Delay call]', 'color: red')
-      setTimeout(() => {
-        f(...args)
-      }, amount)
+        if (type !== 'engineInternal') {
+          Stats.gotStat(m, type === 'serverToEngine')
+        }
+      }
+
+      handler(a1, a2, a3, a4, a5)
     }
-  } else {
-    return f
   }
-}
-
-// We basically always delay/log/ensure once all the calls back and forth
-function _wrap<F: Function>(
-  f: F,
-  logType: rpcLogType,
-  amount: number,
-  logTitle: string,
-  logInfo?: Object
-): F {
-  const logged: F = _makeLogged(f, logType, logTitle, logInfo)
-  const delayed: F = _makeDelayed(logged, amount)
-  const onceOnly: F = _makeOnceOnly(delayed)
-  return onceOnly
+  return wrapped
 }
 
 // Logging for rpcs
-function rpcLog(type: rpcLogType, title: string, info?: Object): void {
+function rpcLog(info: {method: string, reason?: string, extra?: Object, type: string}): void {
   if (!printRPC) {
     return
   }
 
   const prefix = {
-    engineInternal: '[engine]',
-    engineToServer: '[engine] ->',
-    serverToEngine: '[engine] <-',
-  }[type]
-  const style = {
-    engineInternal: 'color: purple',
-    engineToServer: 'color: blue',
-    serverToEngine: 'color: green',
-  }[type]
+    engineInternal: '🏎 ℹ️',
+    engineToServer: '🏎 ↗️',
+    serverToEngine: '🏎 ⤵️',
+  }[info.type]
 
   requestIdleCallback(
     () => {
-      localLog(`%c${prefix}`, style, title, info)
+      const params = [info.reason, info.method, info.extra].filter(Boolean)
+      localLog(prefix, ...params)
     },
     {timeout: 1e3}
   )
 }
 
 class TransportShared extends RobustTransport {
-  // $FlowIssue
-  constructor(opts, connectCallback, disconnectCallback, incomingRPCCallback, writeCallback) {
-    const hooks = {
+  constructor(
+    opts: Object,
+    connectCallback: () => void,
+    disconnectCallback: () => void,
+    incomingRPCCallback: (a: any) => void,
+    writeCallback: any
+  ) {
+    super(opts)
+
+    this.hooks = {
       connected: () => {
-        // $FlowIssue complains that this might be null
```

Comment:
flow rightly complains about using this before super is called. turns out we can just set hooks after